js/ui/hierarchical_collection/ui.data_adapter.js

```diff
@@ -419,16 +421,28 @@ var DataAdapter = Class.inherit({
         return !this.options.searchValue.length || !!this._filterDataStructure(this.options.searchValue, [item]).length;
     },
 
+    _createCriteria: function(selector, value, op) {
+        var searchFilter = [];
+        if(!Array.isArray(selector)) {
+            return [selector, op, value];
+        }
+        iteratorUtils.each(selector, function(i, item) {
+            if(searchFilter.length) {
+                searchFilter.push("or");
+            }
+            searchFilter.push([item, op, value]);
```

Comment:
NB: If case with combined filters occurs more often, it would be better rid of double push operation

```
iteratorUtils.each(selector, function(i, item) {
  searchFilter.push([item, op, value], "or");
}
searchFilter.pop();